src/object.cr

```diff
@@ -166,7 +166,7 @@ class Object
   end
 
   # Returns a pretty printed version of `self`.
-  def pretty_inspect(width = 79, newline = "\n", indent = 0) : String
+  def pretty_inspect(width : Int = 79, newline = "\n", indent : Int = 0) : String
```

Comment:
Ditto

> This kind of restriction has a different meaning with the released compiler and compiled compiler

But since this is a public method, this will restrict from any int ot just DefaultInt eventually. Is that intended?

Reply:
Yes, it's intended. The idea is that only `Int` is used in public APIs because everyone will just use `Int`, everywhere. In the rare case where you need another int type, if it's `Int32` or less then autocasting will automatically work, and if not you'll need to call `to_i` if needed.

I was commenting to @waj that if all methods in the std expect just `Int` there's less code generated: just one overload for `Int`. Right now if you index `Array` with `Int8` you get an extra overload and that's just more boilerplate code, bigger binaries and slower compilation.